### ISSUE TITLE
Support MutiGPUs or multi nodes?

### ISSUE THREAD

**Issue by PoisonKiller** [Lightricks/LTX-Video-Trainer#20]
I want to know can I use multi gpus to train lora-version or SFT ltxv-0.9.7-13b model?

**Comment by matanby** [comment]
(#23)
There is a WIP [branch](https://github.com/Lightricks/LTX-Video-Trainer/tree/multi-gpu-support) that implements multi-GPU training support.
We hope to rebase and merge it soon, but you can try to switch over and use it.

**Comment by StarsTesla** [comment]
@matanby It is a need for support FSDP or Deepspeed, make param on multi gpu/nodes, I tried 2b/13b full model training on 48GB L20 GPU, OOM! Well lora is fine, but sometimes is not enough.

**Comment by matanby** [comment]
I believe the 2B model can fit entirely on a GPU with 40GB of VRAM, depending on the sequence length used when training the model, obviously.

The 13B model will need to be sharded across multiple GPUs for a full-rank training without gradient checkpointing, quantization, etc.

**Comment by StarsTesla** [comment]
@matanby This is weird, Here I give my training config, I directly run this on multi-gpu training of 8 L20 for 48GB:
```yaml
# LTXV FULL Training Configuration

# Model configuration
model:
  # model_source: "LTXV_13B_097_DEV" # Options: "LTXV_13B_097_DEV", "LTXV_2B_0.9.6_DEV", "LTXV_2B_0.9.5", "LTXV_2B_0.9.1", "LTXV_2B_0.9.0", or a HF repo/local path
  model_source: "ltxv-2b-0.9.6-dev-04-25.safetensors"
  training_mode: "full" # Options: "lora" or "full"
  load_checkpoint: null # Path to checkpoint file or directory to resume from. If directory, latest checkpoint will be used.


# Optimization configuration
optimization:
  learning_rate: 1e-5
  steps: 2000
  batch_size: 1
  gradient_accumulation_steps: 1
  max_grad_norm: 1.0
  optimizer_type: "adamw" # Options: "adamw" or "adamw8bit"
  scheduler_type: "linear" # Options: "constant", "linear", "cosine", "cosine_with_restarts", "polynomial"
  scheduler_params: {}
  enable_gradient_checkpointing: false
  first_frame_conditioning_p: 0.1

# Acceleration optimization
acceleration:
  mixed_precision_mode: "bf16" # Options: "no", "fp16", "bf16"
  quantization: null # Options: null, "int8-quanto", "int4-quanto", "int2-quanto", "fp8-quanto", "fp8uz-quanto"
  load_text_encoder_in_8bit: true # Load text encoder in 8-bit precision to save memory
  compile_with_inductor: true
  compilation_mode: "reduce-overhead" # Options: "default", "reduce-overhead", "max-autotune"

# Data configuration
data:
  preprocessed_data_root: "waymo_LTX/pinhole_front"
  num_dataloader_workers: 2

# Validation configuration
validation:
  prompts:
    - "The video is captured from a camera mounted on a car. The camera is facing forward. The video is taken from the perspective of a vehicle's dashboard camera. The video captures a rainy day on a street in San Francisco. The street is lined with parked cars, including a silver SUV and a red car. The rain is falling heavily, creating a misty atmosphere. The street is wet, and the raindrops can be seen falling on the windshield of the car from which the video is taken. The street is surrounded by buildings and trees, and the hills in the background are visible through the mist. The video is taken from the perspective of a driver, and the windshield wipers can be seen in motion. The overall mood of the video is gloomy and atmospheric, with the rain and mist creating a sense of isolation and melancholy."
    - "The video is captured from a camera mounted on a car. The camera is facing forward. The video is taken from the perspective of a vehicle's dashboard camera. The video captures a rainy day in a residential neighborhood. The street is slick with rain, and the sky is a uniform gray, suggesting a heavy downpour. The camera is positioned on the sidewalk, looking down the length of the street.  On the left side of the street, there's a row of houses. The closest one is a single-story building with a flat roof and a white exterior. Further down, the houses become two-story structures with pitched roofs. The houses are painted in various colors, including white, yellow, and red. On the right side of the street, there are several parked cars. The closest one is a silver SUV, followed by a red car, a blue car, and another red car. The cars are parked parallel to the curb. The street itself is lined with trees, and there are power lines running along the right side of the street. In the distance, you can see a hill dotted with houses, and the tops of trees are visible above the houses. The video is taken from a low angle, which emphasizes the slope of the street and the hill in the background. The rain is falling heavily, and the street is glistening with wetness. The overall atmosphere is one of a quiet, rainy day in a suburban neighborhood."
    - "The video is captured from a camera mounted on a car. The camera is facing forward. The video is taken from the perspective of a vehicle's dashboard camera. The video captures a bustling city street, where a red and white articulated bus is the main focus. The bus, labeled 'Clean Air Vehicle', is in motion, driving down the street. The bus is composed of two sections connected by a flexible joint, allowing it to navigate tight turns and narrow streets. The street is lined with tall buildings on both sides, creating a canyon-like effect. The buildings are a mix of modern and older architecture, with some featuring large windows and others having more traditional facades. The street itself is busy with traffic, including cars and pedestrians. In the background, a large bridge stretches across the scene, connecting two parts of the city. The bridge is a prominent feature, adding depth to the urban landscape. The video is taken from a low angle, looking up at the bus and the surrounding buildings. This perspective emphasizes the height of the buildings and the size of the bus. The color palette of the video is dominated by shades of gray, red, and white. The gray tones come from the concrete buildings and the asphalt road, while the red and white are from the bus and some of the buildings. Overall, the video provides a dynamic snapshot of city life, with the 'Clean Air Vehicle' bus serving as a symbol of urban mobility and environmental consciousness."
  negative_prompt: "worst quality, inconsistent motion, blurry, jittery, distorted"
  video_dims: [768, 480, 57] # [width, height, frames]
  seed: 42
  inference_steps: 50
  interval: 250 # Set to null to disable validation
  videos_per_prompt: 1
  guidance_scale: 3.5

# Checkpoint configuration
checkpoints:
  interval: 500 # Save a checkpoint every N steps, set to null to disable
  keep_last_n: -1 # Keep only the N most recent checkpoints, set to -1 to keep all

# Flow matching configuration
flow_matching:
  timestep_sampling_mode: "shifted_logit_normal" # Options: "uniform", "shifted_logit_normal"
  timestep_sampling_params: {}

# HuggingFace Hub configuration
hub:
  push_to_hub: false # Whether to push the model weights to the Hugging Face Hub
  hub_model_id: null # Hugging Face Hub repository ID (e.g., 'username/repo-name'). Must be provided if `push_to_hub` is set to True

# General configuration
seed: 42
output_dir: "outputs/waymo_ltxv_2b_full_finetune"


```

**Comment by matanby** [comment]
The key detail is missing here, what sequence length are you training on?

Try to disable the sampling of validation videos, and set `load_text_encoder_in_8bit` to false, to see if this helps.

**Comment by StarsTesla** [comment]
@matanby My understanding is that, the sequence length is calulating by resolution and frames?  Which is 768,x480x 57, then, the text token length should also be counting on? I use VILA to caption, so it would under 256 token for most prompt.
